Layers/Connections/PolygonObjectToRepeatingObject.gd

```diff
@@ -0,0 +1,143 @@
+
+
+func _filter_grid_by_num_neighbors(grid, neighbor_counts, num_neighbors: int):
+	for x in neighbor_counts.keys():
+		for y in neighbor_counts[x].keys():
+			if neighbor_counts[x][y] <= num_neighbors:
+				grid[x].erase(y)
+
+
+func extract_relevant_data(source: LayerComposition, 
+							new_features: Array, 
+							removed_features: Array) -> Variant:
+	# Filter new features for a set "outline" field to create a barrier around it
+	var relevant_new = new_features.filter(func(feature: GeoFeature): 
+				return feature.get_attribute("outline") != "")
+	var relevant_old = removed_features.filter(func(feature: GeoFeature): 
+				return feature.get_attribute("outline") != "")
+	return {"new": relevant_new, "removed": relevant_old}
+
+
+func apply_to_target(target: LayerComposition, features: Variant):
+	var new_features = features["new"]
+	var removed_features = features["removed"]
+	
+	# Remove all outdated features
+	for feature in removed_features:
+		var line_layer: GeoFeatureLayer = target.render_info.geo_feature_layer
+		var geo_line = instanced_geo_lines[feature.get_id()]
+		line_layer.remove_feature(geo_line)
+	
+	# Create new features
+	for feature in new_features:
+		# Obtain the source composition parent node for the feature
+		var parent = source_composition.render_info.render_scene.get_node(
+			var_to_str(feature.get_id()))
+		# Obtain all positions of the set objects in this parent node
+		var node_positions = parent.get_children().map(
+			func(c: Node): return Vector2(c.position.x, c.position.z))
+		
+		# Statically typed arrays in godot require this
+		var node_positions_typed: Array[Vector2]
+		node_positions_typed.assign(node_positions)
+		# TODO: try large areas and see whether filtering the nodes improves performance
+		# TODO: otherwise we might also remove the node-naming in RepeatingObjectRenderer.spiral()
+		#var node_names = parent.get_children().map(func(c: Node): return c.name)
+		#var grid = _build_grid_from_names(node_names)
+		#var neighbor_counts = _count_neighbors(grid)
+		#_filter_grid_by_num_neighbors(grid, neighbor_counts, 4)
+		#var relevant_positions = []
+		#for x in grid.keys():
+			#for y in grid[x].keys():
+				#pass#relevant_positions.append(parent.get_node("%f_%f" % [x, y]).position)
+		
+		# Calculate concave hull boundary
+		var hulls = ConcaveHull.get_hulls(node_positions_typed)
+		
+		var line_layer: GeoFeatureLayer = target.render_info.geo_feature_layer
+		var height_layer: GeoRasterLayer = target.render_info.ground_height_layer
+		var center = target.render_info.render_scene.center
+		
+		for hull in hulls:
+			# Offset the boundary so it does not intersect with the objects
+			var directions = GeometryUtil.get_polygon_vertex_directions(hull)
+			var offset_verts = GeometryUtil.offset_polygon_vertices(
+				hull, directions, -source_composition.render_info.render_scene.offset)
+			
+			# Create geoline and its underlying curve3d
+			var curve := Curve3D.new()
+			curve.bake_interval = 0.
```

Comment:
I can't find anything in the Godot docs about setting this to 0, they only say:

> The distance in meters between two adjacent cached points. Changing it forces the cache to be recomputed the next time the [get_baked_points()](https://docs.godotengine.org/en/stable/classes/class_curve3d.html#class-curve3d-method-get-baked-points) or [get_baked_length()](https://docs.godotengine.org/en/stable/classes/class_curve3d.html#class-curve3d-method-get-baked-length) function is called. The smaller the distance, the more points in the cache and the more memory it will consume, so use with care.

feels like setting it to 0 would result in an endless loop in the engine, is that some loophole we use to avoid baking anything? if so, add a comment